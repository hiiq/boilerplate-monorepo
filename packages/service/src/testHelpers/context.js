import { User, Utils } from '@boilerplate-monorepo/common';

const stubLoader = name => ({
  clear: td.func(`${name}.clear`),
  load: td.func(`${name}.load`),
});

const db = {
  user: {
    read: jest.fn().mockName('read'),
    readRaw: jest.fn().mockName('readRaw'),
    recoveryFind: jest.fn().mockName('recoveryFind'),
    register: jest.fn().mockName('register'),
    revokeRefreshTokens: jest.fn().mockName('revokeRefreshTokens'),
    save: jest.fn().mockName('save'),
  },
  users: {
    read: jest.fn().mockName('read'),
    readPaginated: jest.fn().mockName('readPaginated'),
    readRaw: jest.fn().mockName('readRaw'),
  },
};

const make = Utils.makeSafeExample({
  db,
  ip: '127.0.0.1',
  loaders: {
    user: stubLoader('user'),
  },
  req: {
    headers: {},
    ip: '127.0.0.1',
  },
  res: {
    cookie: jest.fn().mockName('cookie'),
  },
  user: User.apiExample(),
});

export { make };
