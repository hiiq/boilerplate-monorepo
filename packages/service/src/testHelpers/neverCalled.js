const neverCalled = { ignoreExtraArgs: true, times: 0 };

export { neverCalled };
