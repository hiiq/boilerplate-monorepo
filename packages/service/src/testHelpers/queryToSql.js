import { pick } from 'ramda';

const queryToSql = pick(['bindings', 'sql', 'transaction']);

export { queryToSql };
