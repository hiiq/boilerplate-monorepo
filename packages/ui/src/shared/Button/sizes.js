import { oneOf } from 'prop-types';
import { values } from 'ramda';
import { css } from 'styled-components/macro';
import { Layout } from 'types/layout';

const Size = {
  DEFAULT: 'default',
  LARGE: 'large',
  SMALL: 'small',
};

const styles = css`
  ${({ size = Size.DEFAULT }) => {
    switch (size) {
      case Size.LARGE:
        return css`
          font-size: ${1.125 * Layout.BASE_UNIT}rem;
          padding: ${Layout.BASE_UNIT}rem;
        `;
      case Size.SMALL:
        return css`
          font-size: ${0.875 * Layout.BASE_UNIT}rem;
          padding: ${0.25 * Layout.BASE_UNIT}rem;
        `;
      default:
        return css`
          font-size: ${Layout.BASE_UNIT}rem;
          padding: ${0.5 * Layout.BASE_UNIT}rem;
        `;
    }
  }}
`;

const propTypes = oneOf(values(Size));

export { Size, propTypes, styles };
