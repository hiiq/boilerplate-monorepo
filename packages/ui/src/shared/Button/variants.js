import { darken, lighten } from 'polished';
import { oneOf } from 'prop-types';
import { values } from 'ramda';
import { css } from 'styled-components/macro';
import { variants } from 'styled-theming';
import { Color } from 'types/color';
import { DisplayMode } from 'types/displayMode';

const contextProp = 'context';
const themeProp = 'mode';

const Context = {
  DEFAULT: 'default',
  MENU_ITEM: 'menuItem',
  PRIMARY: 'primary',
};

const styles = variants(themeProp, contextProp, {
  [Context.DEFAULT]: {
    [DisplayMode.DARK]: css`
      background-color: ${Color.white};
      border-color: ${lighten(0.2, Color.white)};
      color: ${Color.black};
    `,
    [DisplayMode.LIGHT]: css`
      background-color: ${Color.black};
      border-color: ${darken(0.2, Color.black)};
      color: ${Color.white};
    `,
  },
  [Context.MENU_ITEM]: {
    [DisplayMode.DARK]: css`
      background-color: ${Color.white};
      border-color: transparent;
      color: ${Color.primary};
    `,
    [DisplayMode.LIGHT]: css`
      background-color: ${Color.primary};
      border-color: transparent;
      color: ${Color.white};
    `,
  },
  [Context.PRIMARY]: {
    [DisplayMode.DARK]: css`
      background-color: ${Color.white};
      border-color: ${Color.primary};
      color: ${Color.primary};
    `,
    [DisplayMode.LIGHT]: css`
      background-color: ${Color.primary};
      border-color: ${darken(0.2, Color.primary)};
      color: ${Color.white};
    `,
  },
});

const propTypes = oneOf(values(Context));

export { Context, propTypes, styles };
