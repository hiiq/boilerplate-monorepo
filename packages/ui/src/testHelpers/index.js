import { render } from '@testing-library/react';
import { node } from 'prop-types';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components/macro';
import { Theme } from 'types/theme';

export * from '@testing-library/react';

jest.mock('shared/useTheme');
jest.mock('shared/useTranslate');

const AllTheProviders = ({ children }) => (
  <Router>
    <ThemeProvider theme={Theme.example()}>{children}</ThemeProvider>
  </Router>
);

AllTheProviders.propTypes = {
  children: node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export { customRender as render };
