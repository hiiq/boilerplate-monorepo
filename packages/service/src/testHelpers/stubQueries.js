import knexMock from 'mock-knex';
import { queryToSql } from './queryToSql';

const stubQueries = (mutateFn, responses = []) => {
  const tracker = knexMock.getTracker();

  tracker.install();

  let queries = [];

  tracker.on('query', (query, step) => {
    queries = [...queries, queryToSql(query)];
    mutateFn(queries);
    query.response(responses[step - 1]);
  });
};

export { stubQueries };
