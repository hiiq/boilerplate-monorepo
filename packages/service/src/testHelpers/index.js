import { make as makeContext } from './context';
import {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
} from './graphql';
import { neverCalled } from './neverCalled';
import { queryToSql } from './queryToSql';
import { stubQueries } from './stubQueries';

export {
  graphqlClient,
  makeContext,
  makeGraphqlClient,
  neverCalled,
  queryToSql,
  responseToData,
  responseToErrorCode,
  stubQueries,
};
