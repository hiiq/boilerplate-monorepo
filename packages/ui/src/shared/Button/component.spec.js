import React from 'react';
import { render } from 'testHelpers';
import { Button, domTestId } from '.';
import { Size } from './sizes';
import { Context } from './variants';

describe('<Button/>', () => {
  it('renders children properly', () => {
    const defaultProps = { children: <x-child />, label: 'LABEL' };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders primary context properly', () => {
    const defaultProps = {
      context: Context.PRIMARY,
      label: 'LABEL',
      text: 'TEXT',
    };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders menu item context properly', () => {
    const defaultProps = {
      context: Context.MENU_ITEM,
      label: 'LABEL',
      text: 'TEXT',
    };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders small size context properly', () => {
    const defaultProps = {
      size: Size.SMALL,
      text: 'TEXT',
    };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders default size context properly', () => {
    const defaultProps = {
      size: Size.DEFAULT,
      text: 'TEXT',
    };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });

  it('renders large size context properly', () => {
    const defaultProps = {
      size: Size.LARGE,
      text: 'TEXT',
    };
    const { getByTestId } = render(<Button {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });
});
