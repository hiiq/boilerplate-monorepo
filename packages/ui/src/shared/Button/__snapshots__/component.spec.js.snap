// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders children properly 1`] = `
<button
  aria-disabled="false"
  aria-label="LABEL"
  class="Button__Styled-sc-13kes55-0 dVNaCg"
  data-testid="Button"
  type="button"
>
  <x-child />
</button>
`;

exports[`<Button/> renders default size context properly 1`] = `
<button
  aria-disabled="false"
  aria-label="TEXT"
  class="Button__Styled-sc-13kes55-0 dVNaCg"
  data-testid="Button"
  type="button"
>
  TEXT
</button>
`;

exports[`<Button/> renders large size context properly 1`] = `
<button
  aria-disabled="false"
  aria-label="TEXT"
  class="Button__Styled-sc-13kes55-0 eakAET"
  data-testid="Button"
  type="button"
>
  TEXT
</button>
`;

exports[`<Button/> renders menu item context properly 1`] = `
<button
  aria-disabled="false"
  aria-label="LABEL"
  class="Button__Styled-sc-13kes55-0 enGHiC"
  data-testid="Button"
  type="button"
>
  TEXT
</button>
`;

exports[`<Button/> renders primary context properly 1`] = `
<button
  aria-disabled="false"
  aria-label="LABEL"
  class="Button__Styled-sc-13kes55-0 fgcjcs"
  data-testid="Button"
  type="button"
>
  TEXT
</button>
`;

exports[`<Button/> renders small size context properly 1`] = `
<button
  aria-disabled="false"
  aria-label="TEXT"
  class="Button__Styled-sc-13kes55-0 jTnnQd"
  data-testid="Button"
  type="button"
>
  TEXT
</button>
`;
